dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -315,4 +339,47 @@ public int size() {
             throw new UnsupportedOperationException("size should never be called on a ConcurrentLinkedDeque");
         }
     }
+
+    interface QueuedChannelInstrumentation {
+        Counter requestsQueued();
+
+        Timer requestQueuedTime();
+    }
+
+    static QueuedChannelInstrumentation channelInstrumentation(DialogueClientMetrics metrics, String channelName) {
+        return new QueuedChannelInstrumentation() {
+            @Override
+            public Counter requestsQueued() {
+                return metrics.requestsQueued(channelName);
+            }
+
+            @Override
+            public Timer requestQueuedTime() {
+                return metrics.requestQueuedTime(channelName);
+            }
+        };
+    }
+
+    static QueuedChannelInstrumentation endpointInstrumentation(
+            DialogueClientMetrics metrics, String channelName, String service, String endpoint) {
```

Comment:
Three strings in a row makes me nervous, what about just passing in the entire `com.palantir.dialogue.Endpoint` instead of the last two args, and then pulling out endpoint.servicveName() and endpoint.endpointName() as necessary below?

Reply:
We can do that, I wanted to be careful not to hold an instance of the endpoint and potentially leak temporary instances.